Feature/#120 [Order] 주문 검색 api, 동적 쿼리 작성

jin2304 opened the pull request:
## #️⃣연관된 이슈
- 이슈 번호: #120 

## 변경 타입
- [x] 신규 기능 추가/수정

## 변경 내용
 -  **접근 권한 관리** 로직 추가
    - **고객:** 자신의 주문 내역만 조회 가능
    - **가게 주인:** 자신의 가게 주문 내역 조회 가능
    - **관리자:** 모든 가게 및 주문 내역 조회 가능

- **to-be**(변경 후 설명을 여기에 작성)
  - 주문 검색 api 추가
  - 주문 검색 api 동적 쿼리 추가
  - 주문 검색 api HTTP 테스트


## 체크리스트
- [x] 코드가 제대로 동작하는지 확인했습니다.

jin2304 committed:
[#120] feat: 주문 검색 api 추가

jin2304 committed:
[#120] feat: 주문 검색 api 동적 쿼리 추가

jin2304 committed:
[#120] feat: 주문 검색 api HTTP 테스트

jin2304 committed:
Merge branch 'dev' of https://github.com/toyland/toyland-project into feature/#120

hanjihoon03 reviewed (commented):
수고했어요~  요구사항 수정해야해요! 주문 생성 후 5분 이내에만 취소 가능, 사용자가 주문시 배송지 정보 등록, 주문시 요청 사항도 저장해야 해요!

jin2304 committed:
[#120] refactor: 주문 검색 api 수정

-데이터를 요청할 때, JSON 형식이 아닌 쿼리 파라미터로 전송

jin2304 committed:
Merge branch 'dev' of https://github.com/toyland/toyland-project into feature/#120

jin2304 committed:
[#120] refactor: 에러처리 리팩터링

jin2304 committed:
[#120] refactor: 주문 검색 api 수정

jin2304 committed:
[#120] refactor: 주문 검색 동적 쿼리 수정

-고객 입장에서 자신의 주문 목록을 조회할 때 사용하는 loginUserId와
-관리자 입장에서 특정 회원의 주문 목록을 조회할 때 사용하는 userId로 분리

jin2304 committed:
[#120] feat: 주문 검색 api HTTP 테스트 추가

jin2304 committed:
Merge branch 'dev' of https://github.com/toyland/toyland-project into feature/#120

letsgilit commented:
고생하셨습니다.
